Properly validate localpart when parsing UserId (#799)
ruma/ruma: crates/ruma-identifiers/CHANGELOG.md

* Rename `RoomVersionId::Version{X}` variants to `RoomVersionId::V{X}`
* Rename `RoomIdOrAliasId` to `RoomOrAliasId`

Bug fixes:

> * Properly validate localpart when building a `UserId` via `parse_with_server_name` (and it's variants). This function was accepting localparts which were violating the `localpart_is_fully_conforming` validation. If an invalid user was built, it would panic when calling to `is_historical`.

Reviewer: Can you rewrap this at 100 chars like above? Also there's a small typo:

```suggestion
* Properly validate localpart when building a `UserId` via `parse_with_server_name` (and its variants). This function was accepting localparts which were violating the `localpart_is_fully_conforming` validation. If an invalid user was built, it would panic when calling `is_historical`.
```
